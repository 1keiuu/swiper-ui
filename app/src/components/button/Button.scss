.button {
  height: 60px;
  width: 60px;
  border-radius: 50%;
  background: white;
  box-shadow: 3px 3px 20px -6px rgba(0, 0, 0, 0.2);
  display: flex;
  justify-content: center;
  align-items: center;
  border: 0;
  cursor: pointer;
  img {
    width: 50%;
  }
}
