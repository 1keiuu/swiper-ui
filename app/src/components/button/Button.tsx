import React, { useContext, useState } from "react";
import "./Button.scss";
import { currentCardStore } from "../../store/currentCard";

type ButtonProps = {
  status: string;
};

const handleClick = (status: string, dispatch: ({ type: string }) => {}) => {
  dispatch({ type: "INCREMENT_INDEX" });
  dispatch({ type: "CHANGE_STATUS", status: status });
};

const Button: React.FC<ButtonProps> = (props) => {
  const { state, dispatch } = useContext(currentCardStore);
  switch (props.status) {
    case "like":
      return (
        <button
          className="button"
          onClick={() => {
            handleClick(props.status, dispatch);
          }}
        >
          <img src={`../../assets/heart.png`} />
        </button>
      );
    case "nope":
      return (
        <button
          className="button"
          onClick={() => {
            handleClick(props.status, dispatch);
          }}
        >
          <img src={`../../assets/close.png`} />
        </button>
      );
  }
};

export default Button;
