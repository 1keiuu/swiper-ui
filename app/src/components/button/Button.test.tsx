import React from "react";
import { cleanup, fireEvent, render, screen } from "@testing-library/react";

import Button from "./Button";

describe("Button", () => {
  afterEach(() => {
    cleanup();
  });
  test("render as before.", () => {
    const { asFragment } = render(<Button buttonType="like" isEmpty={false} />);
    expect(asFragment()).toMatchSnapshot();
  });
  test("is active when isEmpty is false.", () => {
    const rendered = render(<Button buttonType="like" isEmpty={false} />);
    expect(rendered.container.firstChild).toBeEnabled;
  });
  test("is disabled when isEmpty is true.", () => {
    const rendered = render(<Button buttonType="like" isEmpty={true} />);
    expect(rendered.container.firstChild).toBeDisabled;
  });
});
