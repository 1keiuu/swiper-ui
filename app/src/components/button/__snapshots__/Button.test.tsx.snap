// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render 1`] = `
<DocumentFragment>
  <button
    class="button --like"
  >
    <img
      src="test-file-stub"
    />
  </button>
</DocumentFragment>
`;
